src/sentry/static/sentry/app/components/editableText.tsx

```diff
@@ -0,0 +1,191 @@
+import Field from 'app/views/settings/components/forms/field';
+
+type Props = {
+  value: string;
+  onChange: (value: string) => void;
+};
+
+function EditableText({value, onChange}: Props) {
+  const [isEditing, setIsEditing] = useState(false);
+  const [inputValue, setInputValue] = useState(value);
+
+  const isEmpty = !inputValue.trim();
+
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const contentRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const enter = useKeypress('Enter');
+  const esc = useKeypress('Escape');
+
+  // check to see if the user clicked outside of this component
+  useOnClickOutside(wrapperRef, () => {
+    if (isEditing && !isEmpty) {
+      onChange(inputValue);
+      setIsEditing(false);
+    }
+  });
+
+  const onEnter = useCallback(() => {
+    if (enter && !isEmpty) {
+      onChange(inputValue);
+      setIsEditing(false);
+    }
+  }, [enter, inputValue, onChange]);
+
+  const onEsc = useCallback(() => {
+    if (esc) {
+      setInputValue(value);
+      setIsEditing(false);
+    }
+  }, [esc, value]);
+
+  // focus the cursor in the input field on edit start
+  useEffect(() => {
+    if (isEditing) {
+      const inputElement = inputRef.current;
+      if (defined(inputElement)) {
+        inputElement.focus();
+      }
+    }
+  }, [isEditing]);
+
+  useEffect(() => {
+    if (isEditing) {
+      // if Enter is pressed, save the text and close the editor
+      onEnter();
+      // if Escape is pressed, revert the text and close the editor
+      onEsc();
+    }
+  }, [onEnter, onEsc, isEditing]); // watch the Enter and Escape key presses
+
+  function handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setInputValue(event.target.value);
+  }
+
+  function handleContentClick() {
+    setIsEditing(true);
+  }
+
+  return (
+    <Wrapper ref={wrapperRef}>
+      {isEditing ? (
+        <InputWrapper isEmpty={isEmpty}>
+          <StyledField
+            error={isEmpty ? t('Text required') : undefined}
+            inline={false}
+            flexibleControlStateSize
+            stacked
+            required
+          >
+            <StyledInput ref={inputRef} value={inputValue} onChange={handleInputChange} />
+          </StyledField>
+          <InputLabel>{inputValue}</InputLabel>
+        </InputWrapper>
+      ) : (
+        <Content onClick={handleContentClick} ref={contentRef}>
+          <Label>
+            <InnerLabel>{inputValue}</InnerLabel>
+          </Label>
+          <StyledIconEdit />
+        </Content>
+      )}
+    </Wrapper>
+  );
+}
+
+export default EditableText;
+
+const Content = styled('div')`
+  height: 40px;
+  position: relative;
+  max-width: calc(100% - 22px);
+  padding-right: 22px;
+`;
+
+const Label = styled('div')`
+  display: inline-block;
+  border: 1px solid transparent;
+  border-radius: ${p => p.theme.borderRadius};
+  transition: border 150ms;
+  text-align: left;
+  padding: 0 10px;
```

Comment:
Should this be using a space() value?

Reply:
in this case I had to use 10px...otherwise the input would jump all the time... All css was carefully defined to work well on Safari, Chrome, Firefox and IE Edge